<!WRONG_MODIFIER_TARGET!>data<!> enum class First(val x: Int) {
    A(1),
    B(2)
}

<!WRONG_MODIFIER_TARGET!>data<!> object <!DATA_CLASS_WITHOUT_PARAMETERS!>Second<!>

<!WRONG_MODIFIER_TARGET!>data<!> interface <!DATA_CLASS_WITHOUT_PARAMETERS!>Third<!>

<!WRONG_MODIFIER_TARGET!>data<!> annotation class Fourth(val x: Int)